Error while building

I'm having an issue while running the emulator directly from the docker container.

`docker run -e DISPLAY=$DISPLAY -v /tmp/.X11-unix:/tmp/.X11-unix -v $(pwd)/koreader:/home/ko/koreader -it koreader/koappimage:latest bash -c "source ~/.bashrc && pushd koreader && ./kodev run"`

The previous command stop with
```
install -d thirdparty/lua-Spore/build/x86_64-linux-gnu-debug
rm -f /home/ko/koreader/base/thirdparty/lua-Spore/build/x86_64-linux-gnu-debug/lua-Spore-prefix/src/lua-Spore/../lua-Spore-stamp/lua-Spore-build
rm -f build/x86_64-linux-gnu-debug/rocks/lib/luarocks/rocks/lua-spore/0.3.1-1/lua-spore-0.3.1-1.rockspec
cd thirdparty/lua-Spore/build/x86_64-linux-gnu-debug && \
        cmake  -G Ninja -DOUTPUT_DIR="/home/ko/koreader/base/build/x86_64-linux-gnu-debug" \
        -DLUA_SPORE_VER=0.3.1-1 -DLD="/usr/bin/ccache gcc" \
        -DCC="/usr/bin/ccache gcc" -DCFLAGS="-Og -g -pipe -fno-omit-frame-pointer   -march=native -fPIC -I/home/ko/koreader/base/thirdparty/luajit/build/x86_64-linux-gnu-debug/luajit-prefix/src/luajit/src" \
        -DLUAROCKS="luarocks" \
        -DLUA_INCDIR=""/home/ko/koreader/base/thirdparty/luajit/build/x86_64-linux-gnu-debug/luajit-prefix/src/luajit/src"" \
        -DLUA_LIBDIR=""/home/ko/koreader/base/build/x86_64-linux-gnu-debug/libs/"" \
         \
        /home/ko/koreader/base/thirdparty/lua-Spore && \
        /usr/bin/ninja 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ko/koreader/base/thirdparty/lua-Spore/build/x86_64-linux-gnu-debug
[1/7] Performing download step for 'lua-Spore'
FAILED: lua-Spore-prefix/src/lua-Spore-stamp/lua-Spore-download 
cd /home/ko/koreader/base/thirdparty/lua-Spore/build/x86_64-linux-gnu-debug/lua-Spore-prefix/src && /usr/bin/cmake -P /home/ko/koreader/base/thirdparty/lua-Spore/build/x86_64-linux-gnu-debug/git_checkout/tmp/lua-Spore-gitclone-tags/0.3.3.cmake && /usr/bin/cmake -E touch /home/ko/koreader/base/thirdparty/lua-Spore/build/x86_64-linux-gnu-debug/lua-Spore-prefix/src/lua-Spore-stamp/lua-Spore-download
Cloning into 'lua-Spore'...
fatal: unable to access 'https://framagit.org/fperrad/lua-Spore/': server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
Cloning into 'lua-Spore'...
fatal: unable to access 'https://framagit.org/fperrad/lua-Spore/': server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
Cloning into 'lua-Spore'...
fatal: unable to access 'https://framagit.org/fperrad/lua-Spore/': server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
-- Had to git clone more than once:
            3 times.
CMake Error at /home/ko/koreader/base/thirdparty/lua-Spore/build/x86_64-linux-gnu-debug/git_checkout/tmp/lua-Spore-gitclone-tags/0.3.3.cmake:66 (message):
  Failed to clone repository: 'https://framagit.org/fperrad/lua-Spore'


ninja: build stopped: subcommand failed.
Makefile.third:678: recipe for target 'build/x86_64-linux-gnu-debug/rocks/lib/luarocks/rocks/lua-spore/0.3.1-1/lua-spore-0.3.1-1.rockspec' failed
make[1]: *** [build/x86_64-linux-gnu-debug/rocks/lib/luarocks/rocks/lua-spore/0.3.1-1/lua-spore-0.3.1-1.rockspec] Error 1
make[1]: Leaving directory '/home/ko/koreader/base'
Makefile:81: recipe for target 'all' failed
make: *** [all] Error 2
Failed to build emulator! Try run with -v for more information.
```
I'm on Ubuntu 20.04.3 using docker 20.10.12, build e91ed57
Trying to open `https://framagit.org/fperrad/lua-Spore/` in my browser show no issue with the certificate.

Latest seems to be outdated. I'll investigate that later (maybe I never set
it up for automated build), but use 0.2.2 instead. See
https://github.com/koreader/virdevenv/pull/68

On Mon, Jan 10, 2022 at 5:33 PM Federico Granata ***@***.***>
wrote:

> I'm having an issue while running the emulator directly from the docker
> container.
>
> docker run -e DISPLAY=$DISPLAY -v /tmp/.X11-unix:/tmp/.X11-unix -v
> $(pwd)/koreader:/home/ko/koreader -it koreader/koappimage:latest bash -c
> "source ~/.bashrc && pushd koreader && ./kodev run"
>
> The previous command stop with
>
> install -d thirdparty/lua-Spore/build/x86_64-linux-gnu-debug
> rm -f /home/ko/koreader/base/thirdparty/lua-Spore/build/x86_64-linux-gnu-debug/lua-Spore-prefix/src/lua-Spore/../lua-Spore-stamp/lua-Spore-build
> rm -f build/x86_64-linux-gnu-debug/rocks/lib/luarocks/rocks/lua-spore/0.3.1-1/lua-spore-0.3.1-1.rockspec
> cd thirdparty/lua-Spore/build/x86_64-linux-gnu-debug && \
>         cmake  -G Ninja -DOUTPUT_DIR="/home/ko/koreader/base/build/x86_64-linux-gnu-debug" \
>         -DLUA_SPORE_VER=0.3.1-1 -DLD="/usr/bin/ccache gcc" \
>         -DCC="/usr/bin/ccache gcc" -DCFLAGS="-Og -g -pipe -fno-omit-frame-pointer   -march=native -fPIC -I/home/ko/koreader/base/thirdparty/luajit/build/x86_64-linux-gnu-debug/luajit-prefix/src/luajit/src" \
>         -DLUAROCKS="luarocks" \
>         -DLUA_INCDIR=""/home/ko/koreader/base/thirdparty/luajit/build/x86_64-linux-gnu-debug/luajit-prefix/src/luajit/src"" \
>         -DLUA_LIBDIR=""/home/ko/koreader/base/build/x86_64-linux-gnu-debug/libs/"" \
>          \
>         /home/ko/koreader/base/thirdparty/lua-Spore && \
>         /usr/bin/ninja
> -- Configuring done
> -- Generating done
> -- Build files have been written to: /home/ko/koreader/base/thirdparty/lua-Spore/build/x86_64-linux-gnu-debug
> [1/7] Performing download step for 'lua-Spore'
> FAILED: lua-Spore-prefix/src/lua-Spore-stamp/lua-Spore-download
> cd /home/ko/koreader/base/thirdparty/lua-Spore/build/x86_64-linux-gnu-debug/lua-Spore-prefix/src && /usr/bin/cmake -P /home/ko/koreader/base/thirdparty/lua-Spore/build/x86_64-linux-gnu-debug/git_checkout/tmp/lua-Spore-gitclone-tags/0.3.3.cmake && /usr/bin/cmake -E touch /home/ko/koreader/base/thirdparty/lua-Spore/build/x86_64-linux-gnu-debug/lua-Spore-prefix/src/lua-Spore-stamp/lua-Spore-download
> Cloning into 'lua-Spore'...
> fatal: unable to access 'https://framagit.org/fperrad/lua-Spore/': server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
> Cloning into 'lua-Spore'...
> fatal: unable to access 'https://framagit.org/fperrad/lua-Spore/': server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
> Cloning into 'lua-Spore'...
> fatal: unable to access 'https://framagit.org/fperrad/lua-Spore/': server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none
> -- Had to git clone more than once:
>             3 times.
> CMake Error at /home/ko/koreader/base/thirdparty/lua-Spore/build/x86_64-linux-gnu-debug/git_checkout/tmp/lua-Spore-gitclone-tags/0.3.3.cmake:66 (message):
>   Failed to clone repository: 'https://framagit.org/fperrad/lua-Spore'
>
>
> ninja: build stopped: subcommand failed.
> Makefile.third:678: recipe for target 'build/x86_64-linux-gnu-debug/rocks/lib/luarocks/rocks/lua-spore/0.3.1-1/lua-spore-0.3.1-1.rockspec' failed
> make[1]: *** [build/x86_64-linux-gnu-debug/rocks/lib/luarocks/rocks/lua-spore/0.3.1-1/lua-spore-0.3.1-1.rockspec] Error 1
> make[1]: Leaving directory '/home/ko/koreader/base'
> Makefile:81: recipe for target 'all' failed
> make: *** [all] Error 2
> Failed to build emulator! Try run with -v for more information.
>
> I'm on Ubuntu 20.04.3 using docker 20.10.12, build e91ed57
> Trying to open https://framagit.org/fperrad/lua-Spore/ in my browser show
> no issue with the certificate.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/koreader/virdevenv/issues/72>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABRQBI643PYX2G67UBTJE3UVMC4XANCNFSM5LUDB3JQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

--- a/README.md
+++ b/README.md
@@ -11,7 +11,9 @@ docker run -v $(pwd)/koreader:/home/ko/koreader -it koreader/koappimage:latest b
 cd koreader && ./kodev fetch-thirdparty
 ```
 
-You can even run the emulator directly from the Docker container provided you have a local X server, more details [here](http://fabiorehm.com/blog/2014/09/11/running-gui-apps-with-docker/). Some further possibilities (e.g., for Mac OS X and Windows) are explored [here](https://stackoverflow.com/questions/16296753/can-you-run-gui-applications-in-a-docker-container). NB The following command assumes KOReader is harmless and partially breaks the regular Docker container isolation.
+You can even run the emulator directly from the Docker container provided you have a local X server, more details [here](http://fabiorehm.com/blog/2014/09/11/running-gui-apps-with-docker/). Some further possibilities (e.g., for Mac OS X and Windows) are explored [here](https://stackoverflow.com/questions/16296753/can-you-run-gui-applications-in-a-docker-container).
+
+**NB The following command assumes KOReader is harmless and partially breaks the regular Docker container isolation. Forwarding X11 is here be dragons territory. Some people have reported segfaults in the emulator when using this method.** You have been warned.
 
 ```
 docker run -e DISPLAY=$DISPLAY -v /tmp/.X11-unix:/tmp/.X11-unix -v $(pwd)/koreader:/home/ko/koreader -it koreader/koappimage:latest bash -c "source ~/.bashrc && pushd koreader && ./kodev run"

